src/AvoidThisInProxy.js

```diff
@@ -0,0 +1,68 @@
+import {
+  getParentOfNodeType,
+  isInSomething,
+  nearestCalleeName,
+} from './lib/utils'
+
+export const MESSAGE_THIS_IN_PROXY = `Avoid using \`this\` in valtio.proxy context.It might lead to unexpected results.
+Using this is valid, but often a pitfall for beginners.`
```

Comment:
```suggestion
Using \`this\` is valid, but often a pitfall for beginners.`
```
Does this work?

Reply:
why I did i not see that....
Yeah works!